prow/gerrit/reporter/reporter.go

```diff
@@ -18,26 +18,79 @@ limitations under the License.
 package reporter
 
 import (
+	"errors"
+	"fmt"
+	"strings"
+
 	"github.com/sirupsen/logrus"
 
 	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/gerrit/client"
 )
 
-// Client holds a gerrit client
+type gerritClient interface {
+	SetReview(instance, id, revision, message string) error
+}
+
+// Client is a gerrit reporter client
 type Client struct {
-	// TODO(krzyzacy): fill this up
-	// gc gerrit.Client
+	gc gerritClient
 }
 
 // NewReporter returns a reporter client
-func NewReporter() *Client {
-	return &Client{}
+func NewReporter(cookiefilePath string, projects map[string][]string) (*Client, error) {
+	gc, err := gerrit.NewClient(projects)
+	if err != nil {
+		return nil, err
+	}
+	gc.Start(cookiefilePath)
+	return &Client{
+		gc: gc,
+	}, nil
+}
+
+// GetName returns the name of the reporter
+func (c *Client) GetName() string {
+	return "gerrit-reporter"
+}
+
+// ShouldReport returns if this prowjob should be reported by the gerrit reporter
+func (c *Client) ShouldReport(pj *v1.ProwJob) bool {
+
+	if pj.Status.State == v1.TriggeredState || pj.Status.State == v1.PendingState {
+		// not done yet
+		return false
+	}
+
+	// has gerrit metadata (scheduled by gerrit adapter)
+	return pj.ObjectMeta.Annotations["gerrit-id"] != "" &&
+		pj.ObjectMeta.Annotations["gerrit-instance"] != "" &&
+		pj.ObjectMeta.Labels["gerrit-revision"] != ""
 }
 
 // Report will send the current prowjob status as a gerrit review
 func (c *Client) Report(pj *v1.ProwJob) error {
-	logrus.Infof("Reporting %s to gerrit, state %s", pj.Spec.Job, pj.Status.State)
 	// TODO(krzyzacy): we should also do an aggregate report, as golang does in their repo
```

Comment:
is there an issue to track this? if so can we add a ref to it here?

Reply:
opened #9433